util/pingpong.c

```diff
@@ -205,20 +210,52 @@ static const int integ_alphabet_length =
 /*******************************************************************************
  *                                  Compatibility methods
  ******************************************************************************/
-#if defined(__APPLE__) && !HAVE_CLOCK_GETTIME
-int clock_gettime(clockid_t clk_id, struct timespec *tp)
+
+int pp_clock_gettime(struct pp_time *ofit)
 {
+#if defined(__APPLE__)
+	/* clock_gettime(3) portability on OS X / Mac OS X is... complicated:
+	 *
+	 * 1. OS X 10.11 (El Capitan) and below do not have
+	 *    clock_gettime(3).
+	 * 2. MacOS 10.12 (Sierra) and above has clock_gettime(3).
+	 * 3. XCode 8.x allows you to build 10.12 applications on
+	 *    10.11 by setting the env var SDKROOT to point to the
+	 *    10.12 root. On 10.11, the SDK libraries contain a weak
+	 *    symbol for clock_gettime(3) (and some other POSIX
+	 *    functions).  Meaning:
+	 *    1. If you run the resulting executable on 10.11, you get
+	 *       a linker error because there is no strong symbol for
+	 *       clock_gettime behind the weak symbol.
+	 *    2. If you run the resulting executable on 10.12, it
+	 *    works fine.
+	 *
+	 * We *could* have some fairly complicated configury on OS X /
+	 * MacOS, but why bother?  Libfabric is built on OS X / MacOS
+	 * mainly for testing, not high performance -- meaning that
+	 * using gettimeofday(3) is sufficient for timing purposes.
+	 */
 	int retval;
 	struct timeval tv;
 
 	retval = gettimeofday(&tv, NULL);
 
-	tp->tv_sec = tv.tv_sec;
-	tp->tv_nsec = tv.tv_usec * 1000;
+	ofit->sec = tv.tv_sec;
+	ofit->nsec = tv.tv_usec * 1000;
+
+	return retval;
+#else
+	int retval;
+	struct timespec ts;
+
+	retval = clock_gettime(CLOCK_MONOTONIC, &ts);
+
+	ofit->sec = ts.tv_sec;
+	ofit->nsec = ts.tv_nsec;
 
 	return retval;
-}
 #endif
+}
```

Comment:
Unfortunately, we can't use `fi_gettime_*()` here, because those symbols aren't exported from libfabric.la.  However, `fi_gettime_us()` is a trivial 3-line function that is easy to import here and use instead of the existing (more complicated) code.  Done.

As for fabtests, yes, we'll likely have the problem over there, too.  I hadn't looked at that yet.